src/sentry/api/endpoints/dif_files.py

```diff
@@ -6,14 +6,28 @@
 from rest_framework.response import Response
 
 from sentry.utils import json
+from sentry.api.serializers import serialize
 from sentry.api.bases.chunk import ChunkAssembleMixin
 from sentry.api.bases.project import ProjectEndpoint, ProjectReleasePermission
-from sentry.models import File, ChunkFileState
+from sentry.models import File, ChunkFileState, ProjectDSymFile
 
 
 class DifAssembleEndpoint(ChunkAssembleMixin, ProjectEndpoint):
     permission_classes = (ProjectReleasePermission, )
 
+    def _add_project_dsym_to_reponse(self, found_file, response):
+        if found_file is not None:
+            if found_file.headers.get('error', None) is not None:
+                response['error'] = found_file.headers.get('error')
+
+            dsym = ProjectDSymFile.objects.filter(
+                file=found_file
+            ).first()
+            if dsym is not None:
+                response['dif'] = serialize(dsym)
```

Comment:
This should even be required if the state is `OK` and log an error if the `ProjectDsymFile` is missing since it indicates that something went wrong and we weren't able to write the correct state header.

Reply:
We can't check this here, since the original chunked file will be deleted, only the new dsym file is in the database without the `__state` header at all.
So pretty much the API returns `OK` if there is no state header but a file with owned blobs.